src/options.js

```diff
@@ -10,6 +11,18 @@ import { _catchError } from './diff/catch-error';
  * @type {import('./internal').Options}
  */
 const options = {
+	vnode: vnode => {
+		const { type, props } = vnode;
+		// If a Component VNode, check for and apply defaultProps
+		// Note: type may be undefined in development, must never error here.
+		if (typeof type == 'function' && type.defaultProps != null) {
+			for (const i in type.defaultProps) {
+				if (props[i] === UNDEFINED) {
+					props[i] = type.defaultProps[i];
+				}
+			}
+		}
+	},
```

Comment:
wondering if this is the reason for the perf regression. This means we always must call `options.vnode` which we didn't before.

Reply:
It probably is, I did an attempt at making it cheaper by not destructuring but might not help. Was really excited about the byte savings 😅